dss-apps/dss-datamodel-center/dss-datamodel-measure-management/src/main/java/com/webank/wedatasphere/dss/datamodel/measure/service/impl/MeasureServiceImpl.java

```diff
@@ -0,0 +1,214 @@
+import com.webank.wedatasphere.dss.datamodel.measure.dao.DssDatamodelMeasureMapper;
+import com.webank.wedatasphere.dss.datamodel.measure.dto.MeasureQueryDTO;
+import com.webank.wedatasphere.dss.datamodel.measure.entity.DssDatamodelMeasure;
+import com.webank.wedatasphere.dss.datamodel.measure.service.MeasureService;
+import com.webank.wedatasphere.dss.datamodel.measure.vo.MeasureAddVO;
+import com.webank.wedatasphere.dss.datamodel.measure.vo.MeasureEnableVO;
+import com.webank.wedatasphere.dss.datamodel.measure.vo.MeasureQueryVO;
+import com.webank.wedatasphere.dss.datamodel.measure.vo.MeasureUpdateVO;
+import org.apache.linkis.common.exception.ErrorException;
+import org.apache.linkis.server.Message;
+import org.apache.commons.lang3.StringUtils;
+import org.modelmapper.ModelMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import javax.annotation.Resource;
+import java.util.Date;
+import java.util.stream.Collectors;
+
+
+@Service
+public class MeasureServiceImpl extends ServiceImpl<DssDatamodelMeasureMapper, DssDatamodelMeasure> implements MeasureService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MeasureServiceImpl.class);
+
+    private final ModelMapper modelMapper = new ModelMapper();
+
+
+
+    @Resource
+    private AssertsSyncService assertsSyncService;
+
+    @Resource
+    private DatamodelReferencService datamodelReferencService;
+
+    @Override
+    @Transactional(rollbackFor = Exception.class)
+    public long addMeasure(MeasureAddVO vo) throws ErrorException {
+        if (getBaseMapper().selectCount(Wrappers.<DssDatamodelMeasure>lambdaQuery().eq(DssDatamodelMeasure::getName, vo.getName())
+                .or().eq(DssDatamodelMeasure::getFieldIdentifier, vo.getFieldIdentifier())) > 0) {
+            LOGGER.error("errorCode : {}, measure name or field identifier can not repeat, name : {}", ErrorCode.INDICATOR_ADD_ERROR.getCode(), vo.getName());
+            throw new DSSDatamodelCenterException(ErrorCode.MEASURE_ADD_ERROR.getCode(), "measure name or field identifier can not repeat");
+        }
+
+
+        //校验引用情况
+        if(datamodelReferencService.measureReferenceCount(vo.getName())>0
+                ||datamodelReferencService.measureReferenceCount(vo.getFieldIdentifier())>0){
+            LOGGER.error("errorCode : {}, measure name can not be referenced ", ErrorCode.MEASURE_UPDATE_ERROR.getCode());
+            throw new DSSDatamodelCenterException(ErrorCode.MEASURE_UPDATE_ERROR.getCode(), "measure name can not be referenced");
+        }
+
+        DssDatamodelMeasure newOne = modelMapper.map(vo, DssDatamodelMeasure.class);
+        newOne.setCreateTime(new Date());
+        newOne.setUpdateTime(new Date());
+        getBaseMapper().insert(newOne);
+        //同步atlas
+        CreateModelTypeResult result = assertsSyncService.syncCreateModel(
```

Comment:
publisher.publishEvent( new CreateModelEvent(this
                , DataModelSecurityContextHolder.getContext().getDataModelAuthentication().getUser()
                , vo.getFieldIdentifier()
                , ClassificationConstant.MEASURE));